src/forms/stock-dispense/stock-dispense.component.tsx

```diff
@@ -9,12 +9,69 @@ type StockDispenseProps = {
   medicationDispense: MedicationDispense;
   updateInventoryItem: (inventoryItem: InventoryItem) => void;
   inventoryItem: InventoryItem;
+  updateAvailableQuantityToMedicationDispenseQuantity: (inventoryItem: InventoryItem) => void;
 };
 
-const StockDispense: React.FC<StockDispenseProps> = ({ medicationDispense, updateInventoryItem }) => {
+const StockDispense: React.FC<StockDispenseProps> = ({
+  medicationDispense,
+  updateInventoryItem,
+  updateAvailableQuantityToMedicationDispenseQuantity,
+}) => {
   const { t } = useTranslation();
   const drugUuid = medicationDispense?.medicationReference?.reference?.split('/')[1];
   const { inventoryItems, error, isLoading } = useDispenseStock(drugUuid);
+  const validInventoryItems = inventoryItems.filter((item) => isValidBatch(medicationDispense, item));
+
+  function parseDate(dateString) {
+    return new Date(dateString);
+  }
+
+  function isValidBatch(medicationToDispense, inventoryItem) {
+    if (medicationToDispense?.dosageInstruction && medicationToDispense?.dosageInstruction.length > 0) {
+      return medicationToDispense.dosageInstruction.some((instruction) => {
+        if (
+          instruction.timing?.repeat?.duration &&
+          instruction.timing?.repeat?.durationUnit &&
+          inventoryItem.quantity > 0
+        ) {
+          const durationUnit = instruction.timing.repeat.durationUnit;
+          const durationValue = instruction.timing.repeat.duration;
+          const lastMedicationDate = new Date();
+
+          switch (durationUnit) {
+            case 's':
+              lastMedicationDate.setSeconds(lastMedicationDate.getSeconds() + durationValue);
+              break;
+            case 'min':
+              lastMedicationDate.setMinutes(lastMedicationDate.getMinutes() + durationValue);
+              break;
+            case 'h':
+              lastMedicationDate.setHours(lastMedicationDate.getHours() + durationValue);
+              break;
+            case 'd':
+              lastMedicationDate.setDate(lastMedicationDate.getDate() + durationValue);
+              break;
+            case 'wk':
+              lastMedicationDate.setDate(lastMedicationDate.getDate() + durationValue * 7);
+              break;
+            case 'mo':
+              lastMedicationDate.setMonth(lastMedicationDate.getMonth() + durationValue);
+              break;
+            case 'y':
+              lastMedicationDate.setFullYear(lastMedicationDate.getFullYear() + durationValue);
+              break;
+            default:
+              return false;
+          }
+
+          const expiryDate = parseDate(inventoryItem.expiration);
+          return expiryDate > lastMedicationDate;
```

Comment:
Would be good to document what this is testing?  That the drug won't expiry before it is used up?

Reply:
> Would be good to document what this is testing? That the drug won't expiry before it is used up?

yes mark, the intention of the above changes was to ensure that the drugs dispensed to a patient possibly will not expire before the patient finishes his/her medication rather than than dispensing a drug that will expire while the patient is on medication. so its a kinda of prediction by calculating the last date of medication and comparing with the expiry date of the drug

Reply:
It would be good to add a one-line comment to the code explaining this.